src/unstract/sdk/utils/retry_utils.py

```
@@ -0,0 +1,306 @@
+                        attempt, base_delay, multiplier, max_time, jitter
+                    )
+
+                    # Ensure we don't exceed max_time with the delay
+                    remaining_time = max_time - elapsed_time
+                    if delay >= remaining_time:
+                        logger_instance.error(
+                            "Giving up '%s' - next delay %.1fs would exceed max time %.1fs",
+                            func.__name__,
+                            delay,
+                            max_time,
+                        )
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)